test/icinga-legacytimeperiod.cpp

```diff
@@ -213,6 +213,198 @@ BOOST_AUTO_TEST_CASE(simple)
+		BOOST_CHECK_EQUAL(1718236800, segment->Get("end")); // 2024-06-12 24:00:00 UTC
+	}
+	tp->UpdateRegion(1718236800, 1718323200, true);  // 2024-06-13 00:00 - 24:00:00 UTC
+
+	BOOST_CHECK_EQUAL(false, tp->IsInside(1718287200)); // 2024-06-13 14:00:00 UTC
+	{
+		Array::Ptr segments = tp->GetSegments();
+		BOOST_CHECK_EQUAL(0, segments->GetLength()); // There should be no segments at all!
+	}
+
+	// One partially day range shouldn't contain more than a single segment (see https://github.com/Icinga/icinga2/issues/9781).
+	tp->SetRanges(new Dictionary({{"2024-06-12", "10:00-12:00"}}), true);
+	tp->UpdateRegion(1718150400, 1718236800, true);  // 2024-06-12 00:00 - 24:00:00 UTC
+
+	BOOST_CHECK_EQUAL(true, tp->IsInside(1718190000)); // 2024-06-12 11:00:00 UTC
+	{
+		Array::Ptr segments = tp->GetSegments();
+		BOOST_REQUIRE_EQUAL(1, segments->GetLength());
+
+		Dictionary::Ptr segment = segments->Get(0);
+		BOOST_CHECK_EQUAL(1718186400, segment->Get("begin")); // 2024-06-12 10:00:00 UTC (range start date)
+		BOOST_CHECK_EQUAL(1718193600, segment->Get("end")); // 2024-06-12 12:00:00 UTC (range end date)
+	}
+	tp->UpdateRegion(1718236800, 1718323200, true);  // 2024-06-13 00:00 - 24:00:00 UTC
+
+	BOOST_CHECK_EQUAL(false, tp->IsInside(1718287200)); // 2024-06-13 14:00:00 UTC
+	{
+		Array::Ptr segments = tp->GetSegments();
+		BOOST_CHECK_EQUAL(0, segments->GetLength()); // There should be no segments at all!
+	}
+}
+
+BOOST_AUTO_TEST_CASE(include_exclude_timeperiods)
+{
+	Function::Ptr update = new Function("LegacyTimePeriod", LegacyTimePeriod::ScriptFunc, {"tp", "begin", "end"});
+	TimePeriod::Ptr excludedTp = new TimePeriod();
+	excludedTp->SetName("excluded", true);
+	excludedTp->SetUpdate(update, true);
+	excludedTp->SetRanges(new Dictionary({{"2024-06-11", "00:00-24:00"}}), true);
+
+	excludedTp->UpdateRegion(1718064000, 1718323200, true);  // 2024-06-11 00:00 - 2024-06-13 24:00:00 UTC
+
+	BOOST_CHECK_EQUAL(1, excludedTp->GetSegments()->GetLength());
+	BOOST_CHECK_EQUAL(false, excludedTp->IsInside(1718200800)); // 2024-06-12 14:00:00 UTC
+	BOOST_CHECK_EQUAL(false, excludedTp->IsInside(1718287200)); // 2024-06-13 14:00:00 UTC
+
+	// Register the excluded timeperiod to make it globally visible.
+	excludedTp->Register();
+
+	Dictionary::Ptr ranges = new Dictionary({
+		{"2024-06-11", "09:00-17:00"},
+		{"2024-06-12", "09:00-17:00"},
+		{"2024-06-13", "09:00-17:00"}
+	});
+
+	TimePeriod::Ptr tp = new TimePeriod();
+	tp->SetExcludes(new Array({"excluded"}), true);
+	tp->SetUpdate(update, true);
+	tp->SetRanges(ranges, true);
+	tp->UpdateRegion(1718064000, 1718323200, true);  // 2024-06-11 00:00 - 2024-06-13 24:00:00 UTC
+
+	BOOST_CHECK_EQUAL(false, tp->IsInside(1718150400)); // 2024-06-12 00:00:00 UTC
+	BOOST_CHECK_EQUAL(true, tp->IsInside(1718200800)); // 2024-06-12 14:00:00 UTC
+	BOOST_CHECK_EQUAL(false, tp->IsInside(1718323200)); // 2024-06-13 00:00:00 UTC
+	BOOST_CHECK_EQUAL(true, tp->IsInside(1718287200)); // 2024-06-13 14:00:00 UTC
```

Comment:
And this would also be a good place to check that 2024-06-11 14:00:00 UTC is not inside `tp` (as it's in `excludedTp`).